app/models/concerns/host_template_helpers.rb

```diff
@@ -35,6 +35,6 @@ def url_options
   end
 
   def medium_provider
-    @medium_provider ||= Foreman::Plugin.medium_providers.find_provider(@host)
+    @medium_provider ||= Foreman::Plugin.medium_providers.find_provider(@host || self)
```

Comment:
In what case is `@host` present? If the comment on top of this file is correct, this is mixed into host. So `self` should be the only valid option.